File: src/main/java/com/projectx/university/config/RouterConfiguration.java
package com.projectx.university.config;

import com.projectx.university.handler.StudentHandler;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.ServerResponse;

import static org.springframework.web.reactive.function.server.RouterFunctions.route;

@Configuration
public class RouterConfiguration {
    @Bean
    public RouterFunction<ServerResponse> changeTrackingRouter(StudentHandler studentHandler) {
        return route()
                .GET("/api/student/{id}", studentHandler::getStudent)
                .POST("/api/student/", studentHandler::addStudent)
                .build();
    }
}

File: src/main/java/com/projectx/university/config/RouterConfiguration.kt
package com.projectx.university.config

import com.projectx.university.handler.StudentHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.server.RouterFunction
import org.springframework.web.reactive.function.server.RouterFunctions.route
import org.springframework.web.reactive.function.server.ServerResponse

@Configuration
class RouterConfiguration {
    @Bean
    fun studentRouter(studentHandler: StudentHandler): RouterFunction<ServerResponse> =
            route()
                    .GET("/api/student/{id}", studentHandler::getStudent)
                    .POST("/api/student/", studentHandler::addStudent)
                    .build()
}